generators/app/templates/common/.pre-commit-config.yaml

```diff
@@ -63,3 +63,14 @@ repos:
     language: system
     stages: [ commit ]
 <% } -%>
+<% if (includeDvc) { -%>
+- repo: https://github.com/iterative/dvc
```

Comment:
Rather than using git hooks defined in DVC's GitHub (which implies to specify a DVC version, that may diverge from DVC's version on the project), I think it's safer to define our "own" hook, using DVC installation which is already included in the project. Exactly the same as what is done above for `ruff`, `mypy`, etc. And this way you don't need  "additional_dependencies".

It would look like this:

```
      - id: dvc-pre-push
        name: DVC pre-push
        entry: dvc
        args:
          - git-hook
          - pre-push
        require_serial: true
        verbose: true
        stages: [push]
```

Reply:
Not directly related to your contribution, but I just notice the `system` property in the other hooks, which does not seem to be useful... do you know about that one?

Reply:
Your first point about defining hooks is valid to me! I've added `language: system` in the hook for consistency.

I didn't know about the `system` property. After reading the pre-commit documentation landing page, you can run your hook in the language of your choice (python, go) that will create a venv for each tool used for the hook.
At first glance I would say you don't want to use `language: system` as it calls your current development environment that you can mess up with.

Reply:
https://stackoverflow.com/a/70780205 (sorry it's 2023 I still use SO 🤪)

Reply:
Thanks for your response! According to the SO link you've sent, and this [official doc about local hooks](https://pre-commit.com/#repository-local-hooks) (and especially the pylint example), it seems that it does not really matter and we can keep "language: system" for repo local hooks.